client/packages/system/src/Item/Components/StockItemSearchInput/StockItemSearchInput.tsx

```diff
@@ -47,10 +56,15 @@ export const StockItemSearchInput: FC<StockItemSearchInputProps> = ({
     [items]
   );
 
-  const cachedSearchedItems = useMemo(
-    () => ArrayUtils.uniqBy([...items, ...(data?.nodes ?? [])], 'id'),
-    [data]
-  );
+  const cachedSearchedItems = useMemo(() => {
```

Comment:
this is the fix - add the current item to the list, and fetch value from the cached items list.